tools/generator/templates/rules/wscript.jinja2

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2012-2020 Association Prologin <association@prologin.org>
+
+import glob
+import os.path
+
+
+def options(opt):
+    pass
+
+
+def configure(cfg):
+    pass
+
+
+def build(bld):
+    bld.shlib(
+        source = '''
+            {%- for func in game.function %}
+                {%- if func.fct_action %}
+            src/action_{{ func.fct_name }}.cc
+                {%- endif %}
+            {%- endfor %}
+            src/api.cc
+            src/entry.cc
+            src/game_state.cc
+            src/interface.cc
+            src/rules.cc
+        ''',
+        defines = ['MODULE_COLOR=ANSI_COL_BROWN', 'MODULE_NAME="rules"'],
+        target = '{{ game.name }}',
+        use = ['stechec2'],
+    )
+
+    abs_pattern = os.path.join(bld.path.abspath(), 'src/tests/test-*.cc')
+    for test_src in glob.glob(abs_pattern):
+
+        test_name = os.path.split(test_src)[-1]
+        test_name = test_name[5:-3]
```

Comment:
This looks brittle. Can you make it explicit what these lengths are, eg. by using len("the thing")?

Reply:
done